crates/red_knot/src/semantic/types/infer.rs

```diff
@@ -172,9 +172,16 @@ pub fn infer_definition_type(
             let Some(value) = &node.value else {
                 return Ok(Type::Unknown);
             };
-            // TODO handle unpacking assignment correctly (here and for Assignment case, above)
+            // TODO handle unpacking assignment
             infer_expr_type(db, file_id, value)
         }
+        Definition::NamedExpr(node_key) => {
+            let parsed = parse(db.upcast(), file_id)?;
+            let ast = parsed.syntax();
+            let node = node_key.resolve_unwrap(ast.as_any_node_ref());
+            // TODO handle unpacking assignment
```

Comment:
Just realized this comment is wrong/unnecessary, because unpacking assignment can't actually be used in a walrus expression.